event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventJdbcRepository.java

```diff
@@ -198,14 +226,81 @@ protected String jndiName() throws NamingException {
     }
 
     @Override
-    public Stream<Event> getPage(final long offset, final long pageSize,  final Map<String, Object> params) {
+    public Stream<Event> getFeed(final long offset,
+                                 final Link link,
+                                 final long pageSize,
+                                 final Map<String, Object> params) {
+
         try {
-            final Object streamId =  params.get(STREAM_ID);
-            final PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_PAGE);
+            final Object streamId = params.get(STREAM_ID);
+            if (offset == 0L && link.equals(HEAD)) {
```

Comment:
you should always compare from the constant to avoid null pointer, typically safer.
HEAD.equals(link);

Why do we have to check offset is 0L AND the link is head?

Reply:
Not required and removed